source/contributed/typescript-and-screeps-1.md

```diff
@@ -0,0 +1,207 @@
+title: TypeScript and Screeps, from Beginner to Advanced, Pt. 1
+contributed:
+    name: bryan
+    link: https://github.com/bryanbecker
+    date: 2017-05-15
+---
+
+# TypeScript and Screeps, from Beginner to Advanced, Pt. 1
+
+* [Part 1](http://docs.screeps.com/contributed/typescript-and-screeps-1.html)
+**Introduction to TypeScript and setting up a TypeScript + Screeps environment**
+* [Part 2](http://docs.screeps.com/contributed/typescript-and-screeps-2.html)
+In depth look using TypeScript's features to extend the built-in Screeps objects and methods
+* [Part 3](http://docs.screeps.com/contributed/typescript-and-screeps-3.html)
+Getters, setters, and meta programming with decorators
+
+Part One of this series with introduce setting up a TypeScript project for Screeps.  This article will present a basic guide to setting up a TypeScript project for Screeps, and a brief introduction to third party tools.
+
+For users who already have a working TypeScript setup, you can directly skip to [Part 2](http://docs.screeps.com/contributed/typescript-and-screeps-2.html).
+
+For a quick-start, skip down to [screeps-typescript-starter](#screeps-typescript-starter).
+
+## Why TypeScript?
+
+TypeScript is a superset of ECMAScript (a fancy way to say Javascript).  That means it has *all* the features of ECMA, *plus* some extra ones.  When a project is compiled, Typescript will process all your files, and "transpile" them into the target Javascript version (ES6, in the case of Screeps).  This works similarly to Babel, if you have used that in the past.
+
+Using TypeScript can be frustrating when you first start, because it seems the compiler never stops complaining. Power through it. This ends up being the most useful feature of TypeScript once you get going. Towards then, we'll look at some concrete examples of this.
+
+This guide won't attempt to teach you typescript ([Typescript Deep Dive](https://basarat.gitbooks.io/typescript/docs/why-typescript.html) is an execellent starting resource for that), but will provide an introduction in getting set up, and using the more advanced features to your advantage
```

Comment:
> typescript
Use the correct capitalization here (TypeScript).

End the sentence with a full stop.